blog/content/post/2024-03-15-uefi-sb-implementation-mpl.md

```diff
@@ -0,0 +1,385 @@
+    ${sb_menu}=    Enter Secure Boot Menu And Return Construction
+    Enable Secure Boot    ${sb_menu}
+    # Save Changes And Reset
+    # Changes to Secure Boot menu takes action immediately, so we can just reset
+    Reset System
+    Boot Efi File Should Fail    hello.efi    NOT_SIGNED
+```
+
+As you can see, the test case is brief and easily readable thanks to its use of
+predefined keywords.
+
+The test suite can be found in the
+[OSFV repository](https://github.com/Dasharo/open-source-firmware-validation/blob/main/dasharo-security/secure-boot.robot)
+and the detailed description of each test is available on the
+[Unified Test Documentation](https://docs.dasharo.com/unified-test-documentation/dasharo-security/206-secure-boot)
+page.
+
+The test suite can be run using the following command:
+
+```bash
+robot -L TRACE \
+  -v ansible_config:yes \
+  -v rte_ip:<rte_ip> \
+  -v snipeit:no \
+  -v config:mpl-pip4 \
+  -v device_ip:<device_ip> \
+  dasharo-security/secure-boot.robot
+```
+
+To use QEMU execute this instead:
+
+```bash
+robot -L TRACE \
+  -v ansible_config:yes \
+  -v rte_ip:127.0.0.1 \
+  -v snipeit:no \
+  -v config:qemu \
+  -v device_ip:<device_ip> \
+  dasharo-security/secure-boot.robot
+```
+
+### Results
+
+The tests were performed on MPL PIP44. The following images show the test suite
+results:
+![uefi-sb-results-report](/img/uefi-sb-results-pt1.png)
+![uefi-sb-results-report](/img/uefi-sb-results-pt2.png)
+
+The results lead to the following conclusions:
+
+- The state of UEFI Secure Boot functionality can be freely modified from the
+UEFI BIOS Menu, and it is correctly detected from the operating system.
+- Verification of launched images works correctly when UEFI Secure Boot is
+enabled. The firmware allows the execution of files signed with the appropriate
+keys but blocks the booting of unsigned files or files signed with keys not
+present in the database (DB) or files with hashes not stored in the database
+(DB).
+- The firmware correctly recognizes the chain of trust when intermediate
+certificates are used, allowing their use in the verification process.
+- Keys intended for UEFI Secure Boot must be generated using the RSA
+cryptographic algorithm and sizes 2048, 3072, and 4096. Keys generated with the
+ECDSA cryptographic algorithm are not correctly supported.
+- The firmware does not verify the expiration date of certificates during the
+verification of launched files. It means that enrolled certificates may expire
+and will not affect the ability to boot files verified by them.
+- The firmware only allows the resetting of enrolled certificates when they
+have been added from the UEFI BIOS Menu. If they are added from the operating
+system, the certificates are marked as External, and only a full firmware
+reset (for example, by removing the CMOS battery) allows their removal.
+- Certificates enrolled through the Automatic Certificate Provisioning method
+are correctly used to verify launched files.
```

Comment:
This is first time you mention `Automatic Certificate Provisioning` please explain what that means

Reply:
@PLangowski you should explain this, link for the procedure may be also helpful https://github.com/Wind-River/meta-secure-core/tree/master/meta-efi-secure-boot#quick-start-for-the-first-boot

Reply:
I added the explanation 17ef6be357385bbb092bd433a90c0845c7b3b473

Reply:
@PLangowski you did explain it but we should do it the first time given term comes up, now someone will read about this and later it will be explained to them

Reply:
Fixed by adding link to the procedure when the name is first time mentioned in the post

https://github.com/3mdeb/news-and-ideas/pull/590/commits/7426b477df1a3f2f582428d780a871b97af28080#diff-408ae19700697ee529e7794c3865363586a3ffe06152e4c90cb578ba42221b40R151